main.py

```diff
@@ -20,10 +20,22 @@ def sample_top_p(probs: torch.Tensor, p: float):
     return torch.gather(probs_idx, -1, next_token)
 
 
-def sample(logits: torch.Tensor, temperature: float, top_p: float):
-    if temperature > 0:
-        probs = torch.softmax(logits / temperature, dim=-1)
+def sample_top_k(probs: torch.Tensor, top_k: int):
+    probs_sort, probs_idx = torch.sort(probs, dim=-1, descending=True)
+    topk_probs = probs_sort[:top_k]
```

Comment:
I'm not sure this'll work for a batch of sequences. Can you test out if it works?

Reply:
I made a change to support a batch of sequences, it should be ok.